enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
+
+    def my_all?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_any?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
```

Comment:
* [ ] my_any? returns true if the block ever returns a value other than false or nil
  ```ruby
    array = [1,2,5,4,7]
    false_block = proc { |x| x > 10 }
    p array.my_any?(&false_block) == array.any?(&false_block) # false
  ```
* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
  ```ruby
    array = [nil, false, true, []]
    p array.my_any? == array.any? # false
  ```
* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_any?(Integer) == array.any?(Integer) # false
  ```
* [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(/c/) == array.any?(/c/) # false
  ```
* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(5) == array.any?(5) # false
  ```